src/main/kotlin/org/rust/ide/template/postfix/MatchPostfixTemplate.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.template.postfix
+
+import com.intellij.codeInsight.template.postfix.templates.PostfixTemplateWithExpressionSelector
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.ide.inspections.fixes.AddRemainingArmsFix
+import org.rust.ide.inspections.fixes.AddWildcardArmFix
+import org.rust.ide.utils.checkMatch.checkExhaustive
+import org.rust.lang.core.psi.*
+
+class MatchPostfixTemplate(provider: RsPostfixTemplateProvider) :
+    PostfixTemplateWithExpressionSelector(
+        null,
+        "match",
+        "match expr {...}",
+        RsTopMostInScopeSelector(),
+        provider
+    ) {
+
+    override fun expandForChooseExpression(expression: PsiElement, editor: Editor) {
+        if (expression !is RsExpr) return
+
+        val factory = RsPsiFactory(expression.project)
+
+        val project: Project = expression.getProject()
+        val newMatchExpr = factory.tryCreateExpression("match ${expression.text} {\n}") as? RsMatchExpr ?: return
```

Comment:
Do you think we should just do nothing if `tryCreateExpression` returns `null`? IMO it might be better to use `createExpression` to throw an exception: this situation should be very rare, with an opportunity to report an exception to us

Reply:
Good point, `match <expr>` should work always (unless there is a parser bug or something).